source/location_access.cpp

```diff
@@ -2707,4 +2707,30 @@ namespace Exits { //name, scene, hint, events, locations, exits
         }
     }
   }
+
+  //Reset exits and clear items from locations
+  void ResetAllLocations() {
+    for (Exit* exit : allExits) {
+      exit->ResetVariables();
+      for (size_t i = 0; i < exit->locations.size(); i++) {
```

Comment:
This and the following indexing should be able to be compressed to:

```suggestion
      for (auto& locPair : exit->locations) {
```

couldn't it?

Reply:
Indeed it can, ty for pointing it out